Remotion/Web/UnitTests/Core/UI/Controls/Rendering/RendererBaseTest.cs

```diff
@@ -0,0 +1,74 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using JetBrains.Annotations;
+using Moq;
+using NUnit.Framework;
+using Remotion.Globalization;
+using Remotion.Web.UI.Controls;
+using Remotion.Web.UI.Controls.Rendering;
+
+namespace Remotion.Web.UnitTests.Core.UI.Controls.Rendering
+{
+  [TestFixture]
+  public class RendererBaseTest : RendererTestBase
+  {
+    [Test]
+    public void GetResourceManager_DoesNotKeepResourceManagerAlive ()
+    {
+      var renderer = new TestableRendererBase (Mock.Of<IResourceUrlFactory>(), GlobalizationService, Mock.Of<IRenderingFeatures>());
+
+      var resourceManagerWeakReference = CallGetResourceManagerWithManagerAndReturnItsWeakReference(renderer);
+
+      GC.Collect();
+      GC.WaitForPendingFinalizers();
+      GC.Collect();
+
+      Assert.That (resourceManagerWeakReference.IsAlive, Is.False);
+    }
+
+    private static WeakReference CallGetResourceManagerWithManagerAndReturnItsWeakReference (TestableRendererBase renderer)
+    {
+      var resourceManager = ResourceManagerSet.Create (NullResourceManager.Instance);
```

Comment:
Just for consideration (the test would be red anyway, if this changes so there's no hidden test-bug):

There's no contract that ResourcemanagerSet.Create couldn't return the Null-instance for empty sets instead of always creating a set. For the current testcase, a fake IResourceManager would probably be a more robust choice regarding future refactorings.